session.go

```diff
@@ -92,6 +95,7 @@ func newSession(config *Config, conn io.ReadWriteCloser, client bool) *Session {
 		streams:    make(map[uint32]*Stream),
 		synCh:      make(chan struct{}, config.AcceptBacklog),
 		acceptCh:   make(chan *Stream, config.AcceptBacklog),
+		goAwayCh: make(chan struct{}, 1),
```

Comment:
I think we should create the goAwayCh only if the peer is a server so I've moved it to the if client {}else{}. Let me know if you think otherwise. 
   I've noticed that now session_test.go:375 - `client2.GoAway()` always throws `err: session shutdown`. It shouldn't be an issue as the client shouldn't try to open connections after GoAway but still I don't know why it errors out.